students/eowyn/session10/ff-mailroom/ffmailroom.py

```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/env python
+import sys
+import copy
+from textwrap import dedent
+
+
+class Donor():
+    # Keep track of individual donors and their history of donations
+
+    def __init__(self, name):
+        self.donations = []
+        self.name = ''.join(name)  # should we check input?
+
+    def add_donation(self, amt):
+        self.donations.append(float(amt))
+
+    def generate_letter(self):
+        '''
+        Generate a formatted thank you note to a donor and return
+        the string of the thank you note for printing or writing.
+        '''
+        fs = "Thank you, {0}, for your generosity and recent gift of ${1:.2f}."
+        return fs.format(self.name, self.most_recent_donation)
+
+    def mult_donations(self, factor):
+        self.donations = list(map(lambda x: x * factor, self.donations))
+
+    def filter_donations(self, min_donation, max_donation):
+        if min_donation > max_donation:
+            (min_donation, max_donation) = (max_donation, min_donation)
+        self.donations = list(filter(lambda x: x > min_donation and
+                                     x < max_donation, self.donations))
+
+    @property
+    def sum_donations(self):
+        return round(sum(self.donations), 2)
```

Comment:
something to think about -- round() still returns a floating point number, so you still have all the precision (about 15 digits, even though it's rounded.

So ususally we wait for final ouput or printing to round the numbers to even cents.

or:
use integers cents as your unit
or:
use the Decimal data type

Also -- the only operation that might create uneven cents is division (or multiplication by a non-integer...), so if your'e going to round, the avaging function should be the only place you need it.